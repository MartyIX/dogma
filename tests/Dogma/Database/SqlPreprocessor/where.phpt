<?php

/**
 * Test: Nette\Database\SqlPreprocessor: where
 *
 * @author     Vlasta Neubauer
 * @package    Nette\Database
 * @subpackage UnitTests
 */

require_once __DIR__ . '/../connect.inc.php';


$processor = $connection->getSqlPreprocessor();


Assert::same(
	array("UPDATE table WHERE (`a` IS NULL AND `b`=123 AND `c` LIKE 'text' AND `d` IN (123, 456))", array()), 
	$processor->process("UPDATE table WHERE ?", 
		array(
			array(
				'a' => NULL,
				'b' => 123,
				'c' => 'text',
				'd' => array(123, 456))
		)
	)
);

Assert::same(
	array("UPDATE table WHERE (`a` IS NULL AND `b`=123 AND `c` LIKE 'text' AND `d` IN (123, 456))", array()), 
	$processor->process("UPDATE table WHERE ", 
		array(
			array(
				'a' => NULL,
				'b' => 123,
				'c' => 'text',
				'd' => array(123, 456))
		)
	)
);
