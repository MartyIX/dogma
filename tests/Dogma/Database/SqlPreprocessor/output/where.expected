array(
	'UPDATE table WHERE (`a` IS NULL AND `b`=123 AND `c` LIKE \'text\' AND `d` IN (123, 456))',
	array(),
)